client/app/vps/vps-task.service.js

```diff
@@ -0,0 +1,126 @@
+class VpsTaskService {
+    constructor ($http, $q, $rootScope, $timeout, $translate, CloudMessage) {
+        this.$http = $http;
+        this.$q = $q;
+        this.$rootScope = $rootScope;
+
+        this.$timeout = $timeout;
+        this.$translate = $translate;
+        this.CloudMessage = CloudMessage;
+
+        this.firstCall = true;
```

Comment:
I don't like this thing but I understand you have no choice :(

Reply:
Yup. In fact I didn't want to enter $timeout the firstTime if there's no pending task. BUT I want to enter it lastTime to return msg and broadcast it is finish.